airflow-core/src/airflow/ui/src/layouts/Details/DetailsLayout.tsx

```diff
@@ -68,6 +76,61 @@ export const DetailsLayout = ({ children, error, isLoading, tabs }: Props) => {
   const { onClose, onOpen, open } = useDisclosure();
   const [isRightPanelCollapsed, setIsRightPanelCollapsed] = useState(false);
 
+  const getDefaultSizes = useCallback(
+    (): PanelSizes => (dagView === "graph" ? DEFAULT_GRAPH_SIZES : DEFAULT_GRID_SIZES),
+    [dagView],
+  );
+
+  const getSavedSizes = useCallback((): PanelSizes => {
+    const storageKey = `panel-${dagId}-${dagView}`;
+
+    try {
+      const cached = sessionStorage.getItem(storageKey);
+
+      if (cached !== null && cached !== "") {
+        const parsed: unknown = JSON.parse(cached);
+
+        if (isPanelSizes(parsed)) {
+          return parsed;
+        }
+      }
+    } catch {
+      // Failed to parse saved panel sizes - fallback to default
+    }
+
+    return getDefaultSizes();
+  }, [dagId, dagView, getDefaultSizes]);
+
+  const savePanelSizes = useCallback(
+    (sizes: Array<number>) => {
+      try {
+        const storageKey = `panel-${dagId}-${dagView}`;
+
+        sessionStorage.setItem(storageKey, JSON.stringify(sizes));
+      } catch {
+        // Failed to save panel sizes - continue silently
+      }
+    },
+    [dagId, dagView],
+  );
+
+  const [currentSizes, setCurrentSizes] = useState<PanelSizes>(getSavedSizes);
+
+  const [showPanel, setShowPanel] = useState(false);
+
+  useEffect(() => {
+    const newSizes = getSavedSizes();
+
+    setCurrentSizes(newSizes);
+
+    setShowPanel(false);
+    const timer = setTimeout(() => {
+      setShowPanel(true);
+    }, PANEL_RESET_DELAY);
+
+    return () => clearTimeout(timer);
+  }, [dagView, dagId, getSavedSizes]);
```

Comment:
Ideally we would like a solution without useEffect. Is there a way to do it differently ?